controllers/app.js

```diff
@@ -732,6 +735,99 @@ const publicAppDetails = async (req, res) => {
   return res.status(HTTPStatus.OK).json(app)
 }
 
+const uploadMedia = async (req, res) => {
+  if (!req.formdata || !req.formdata.files) {
+    return res.status(HTTPStatus.BAD_REQUEST).send({ errors: ['no files uploaded'] })
+  }
+
+  const files = []
+  const badTypes = []
+  for (const key in req.formdata.files) {
+    const file = req.formdata.files[key]
+    if (file.type.split('/')[0] !== 'image') {
+      badTypes.push(file.name)
+    }
+    files.push(file)
+  }
+
+  if (badTypes.length) {
+    return res.status(HTTPStatus.BAD_REQUEST).send({
+      errors: badTypes.map((f) => ({
+        file: f,
+        error: 'invalid type (image expected)',
+      })),
+    })
+  }
+
+  const app = await models.App.findOne({
+    where: {
+      id: req.params.id,
+      org_id: req.user.org.id,
+      enable: true,
+    },
+  })
+
+  if (!app) return res.status(HTTPStatus.NOT_FOUND).send({ errors: ['App not found'] })
+
+  const storageClient = Storage.getStorageClient()
+  const savePromises = files.map((f) => {
+    const extension = f.name.split('.').pop()
+    return storageClient.saveFile(f.path, `app-media-${req.params.id}-${uuidv4()}.${extension}`)
+  })
+  const saveResults = await Promise.all(savePromises)
+  await Promise.all(files.map((f) => fs.unlink(f.path)))
```

Comment:
Shouldn’t we catch the unlink? Also, why are we unlinking the files? Doesn’t the middleware use a temporary folder for these?

Reply:
There is a global catch around the controller.
We need to remove the files because the middleware lib doesn't (and should not) do that.

Reply:
Handling the unlink could indeed be done to avoid disrupting the flow. Added that.

We need to remove the files because the middleware lib doesn't (and should not) do that.